pkg/kubelet/nodestatus/setters.go

```diff
@@ -90,17 +90,16 @@ func NodeAddress(nodeIP net.IP, // typically Kubelet.nodeIP
 			if nodeIP != nil {
 				enforcedNodeAddresses := []v1.NodeAddress{}
 
-				var nodeIPType v1.NodeAddressType
+				nodeIPTypes := make(map[v1.NodeAddressType]bool)
 				for _, nodeAddress := range nodeAddresses {
 					if nodeAddress.Address == nodeIP.String() {
 						enforcedNodeAddresses = append(enforcedNodeAddresses, v1.NodeAddress{Type: nodeAddress.Type, Address: nodeAddress.Address})
-						nodeIPType = nodeAddress.Type
-						break
+						nodeIPTypes[nodeAddress.Type] = true
 					}
 				}
 				if len(enforcedNodeAddresses) > 0 {
 					for _, nodeAddress := range nodeAddresses {
-						if nodeAddress.Type != nodeIPType && nodeAddress.Type != v1.NodeHostName {
+						if !nodeIPTypes[nodeAddress.Type] && nodeAddress.Type != v1.NodeHostName {
```

Comment:
this panics in case `nodeAddress.Type` key does not exist (e.g. in case `nodeAddress.Address != nodeIP.String()`):
```go
type, exists := nodeIPTypes[nodeAddress.Type]; (!exists || !type)
```

Reply:
Hm? No, reading a non-existent map value is allowed. It just returns the zero value, which does the right thing here. (You're thinking of typecasting, where it *does* panic if you use the 1-return-value form and it fails.)

Reply:
Oh, you are right. Interesting :)